### PR TITLE
Remote Compaction - Remove CFOptions in CompactionServiceOptionsOverride

### PR DESCRIPTION
**Pull request by jaykorean**
# Summary 

In the initial version of the remote compaction, `CompactionServiceOptionsOverride` was introduced temporarily as a workaround to set the pointer configurations explicitly because passing them to the remote worker was not available at that time.

In the Meta's internal Offload infra, we are now able to register these objects set in the CFOptions including CompactionFilterFactory, MergeOperator and others. These pointer configs are passed to the remote worker as `CustomSharedPtr`, `CustomUniquePtr`, or `CustomRawPtr` Option types, which can be created by `Type::createFromString()` on the remote side as long as they are registered in the ObjectRegistry. (As defined in https://github.com/facebook/rocksdb/blob/92ad4a88f3199b013532b37d6598c442319355a5/options/cf_options.cc#L259-L833)

In this PR, we are also not allowing remote compactions with `options.compaction_filter` set. `options.compaction_filter` is serialized as `CustomRawPtr` and `CustomRawPtr` objects are loaded as static objects. This can become a problem if the remote worker processes compactions from multiple different primary hosts with the same compaction filter with different options. To workaround this, we are only allowing `CompactionFilterFactory`. For compactions with `options.compaction_filter` set, they will fall back to local compaction for now.

# Test Plan

Updated `compaction_service_test::CompactionFilter`  test  and added Updated `compaction_service_test::CompactionFilterFactory`
```
./compaction_service_test
```

Also tested with Meta's internal Offload infra

### CODE REVIEW

File: include/rocksdb/options.h [hunk 1]
```diff
@@ -2289,13 +2289,17 @@ struct CompactionServiceOptionsOverride {
   Env* env = Env::Default();
   std::shared_ptr<FileChecksumGenFactory> file_checksum_gen_factory = nullptr;
 
+  // DEPRECATED - CFOption pointer configurations will be passed as static name
+  // These will need to be registered in the remote worker's ObjectRegistry,
+  // then created by Type::CreateFromString() when needed.
```
Comment (cbi42) [inline include/rocksdb/options.h:2294]: This remove compaction APIs are experiment, would it be safe to just remove these?

Comment (jaykorean) [reply include/rocksdb/options.h]: There's an old prototype code in fbcode still referencing this, so that's why I didn't remove them at first, but I guess can clean up the old prototype now before we merge this PR into fbcode.

File: db/db_impl/db_impl_secondary.cc [hunk 2]
```diff
@@ -956,23 +956,10 @@ Status DB::OpenAndCompact(
...
-      override_options.prefix_extractor;
-  compaction_input.column_family.options.table_factory =
-      override_options.table_factory;
-  compaction_input.column_family.options.sst_partitioner_factory =
-      override_options.sst_partitioner_factory;
+  compaction_input.db_options.listeners = override_options.listeners;
+
   compaction_input.column_family.options.table_properties_collector_factories =
       override_options.table_properties_collector_factories;
-  compaction_input.db_options.listeners = override_options.listeners;
```
Comment (anand1976) [inline db/db_impl/db_impl_secondary.cc:975]: Is this related? The `listeners` field is still defined in `CompactionServiceOptionsOverride`.

Comment (jaykorean) [reply db/db_impl/db_impl_secondary.cc:975]: It was moved to line 959

Comment (anand1976) [reply db/db_impl/db_impl_secondary.cc:975]: Sorry, still not fully awake this morning

**Review by anand1976 — APPROVED** [review]

**Comment by jaykorean** [comment]
Plan changed
